src/WebUI/Components/ServicesComponent.tsx

```diff
@@ -47,10 +50,16 @@ export class ServicesComponent extends BaseComponent<IServicesComponentPropertie
         }
     }
 
-    private static _getServiceItems(servicesList: V1ServiceList): IServiceItem[] {
+    private static _getServiceItems(servicesList: V1ServiceList, typeSelections:KubeServiceType[], filterKey:string | undefined): IServiceItem[] {
         let items: IServiceItem[] = [];
 
-        (servicesList && servicesList.items || []).forEach(service => {
+        (servicesList && servicesList.items || []).filter((item) => {

```

Comment:
just a thought, rather doing the filter twice, would it be good to do the filter at once? also should it be good to have a in a separate function?